src/include/duckdb/common/types/row/tuple_data_collection.hpp

```diff
@@ -0,0 +1,235 @@
+	void InitializeAppend(TupleDataPinState &pin_state,
+	                      TupleDataPinProperties = TupleDataPinProperties::UNPIN_AFTER_DONE);
+	//! Initializes the Chunk state of an Append state
+	//! - Useful for optimizing many appends made to the same tuple data collection
+	void InitializeAppend(TupleDataChunkState &chunk_state, vector<column_t> column_ids);
+	//! Append a DataChunk directly to this TupleDataCollection - calls InitializeAppend and Append internally
+	void Append(DataChunk &new_chunk, const SelectionVector &append_sel = *FlatVector::IncrementalSelectionVector(),
+	            idx_t append_count = DConstants::INVALID_INDEX);
+	//! Append a DataChunk directly to this TupleDataCollection - calls InitializeAppend and Append internally
+	void Append(DataChunk &new_chunk, vector<column_t> column_ids,
+	            const SelectionVector &append_sel = *FlatVector::IncrementalSelectionVector(),
+	            const idx_t append_count = DConstants::INVALID_INDEX);
+	//! Append a DataChunk to this TupleDataCollection using the specified append state
+	void Append(TupleDataAppendState &append_state, DataChunk &new_chunk,
+	            const SelectionVector &append_sel = *FlatVector::IncrementalSelectionVector(),
+	            const idx_t append_count = DConstants::INVALID_INDEX);
+	//! Append a DataChunk to this TupleDataCollection using the specified pin and chunk states
+	void Append(TupleDataPinState &pin_state, TupleDataChunkState &chunk_state, DataChunk &new_chunk,
+	            const SelectionVector &append_sel = *FlatVector::IncrementalSelectionVector(),
+	            const idx_t append_count = DConstants::INVALID_INDEX);
+	//! Append a DataChunk to this TupleDataCollection using the specified pin and chunk states
+	//! - ToUnifiedFormat has already been called
+	void AppendUnified(TupleDataPinState &pin_state, TupleDataChunkState &chunk_state, DataChunk &new_chunk,
+	                   const SelectionVector &append_sel = *FlatVector::IncrementalSelectionVector(),
+	                   const idx_t append_count = DConstants::INVALID_INDEX);
+
+	//! Creates a UnifiedVectorFormat in the given chunk state for the given DataChunk
+	static void ToUnifiedFormat(TupleDataChunkState &chunk_state, DataChunk &new_chunk);
+	//! Gets the UnifiedVectorFormat from the chunk state as an array
+	static unique_ptr<UnifiedVectorFormat[]> GetVectorData(TupleDataChunkState &chunk_state);
+	//! Computes the heap sizes for the new DataChunk that will be appended
+	static void ComputeHeapSizes(TupleDataChunkState &chunk_state, DataChunk &new_chunk,
+	                             const SelectionVector &append_sel, const idx_t append_count);
+
+	//! Builds out the buffer space for the specified chunk state
+	void Build(TupleDataPinState &pin_state, TupleDataChunkState &chunk_state, const idx_t append_offset,
+	           const idx_t append_count);
+	//! Scatters the given DataChunk to the rows in the specified append state
```

Comment:
Is there an implied append state?

Reply:
Good catch, this comment is also outdated. An append state is nothing more than:
```c++
struct TupleDataAppendState {
	TupleDataPinState pin_state;
	TupleDataChunkState chunk_state;
};
```
However, in many cases it was useful to pull these apart, and supply the pin/chunk states separately. For example, when appending to a `PartitionedTupleDataCollection`, we need one pin state per partition, but only one chunk state overall.